inc/sitemaps/class-sitemap-cache-data.php

```diff
@@ -0,0 +1,129 @@
+<?php
+/**
+ * @package WPSEO\XML_Sitemaps
+ */
+
+/**
+ * Sitemap Cache Data object, manages sitemap data stored in cache
+ */
+class WPSEO_Sitemap_Cache_Data implements WPSEO_Sitemap_Cache_Data_Interface, Serializable {
+
+	/** @var string Sitemap XML data. */
+	private $sitemap = '';
+
+	/** @var string Status of the sitemap, usable or not. */
+	private $status = self::UNKNOWN;
+
+	/**
+	 * Set the sitemap XML data
+	 *
+	 * @param string $sitemap XML Content of the sitemap.
+	 */
+	public function set_sitemap( $sitemap ) {
+
+		if ( ! is_string( $sitemap ) ) {
+			$sitemap = '';
+		}
+
+		$this->sitemap = $sitemap;
+
+		/**
+		 * Empty sitemap is not usable.
+		 */
+		$this->is_usable( ! empty( $sitemap ) );
+	}
+
+	/**
+	 * Set the status of the sitemap, is it usable.
+	 *
+	 * @param bool|string $valid Is the sitemap valid or not.
+	 *
+	 * @return string The status that was set.
+	 */
+	public function set_status( $valid ) {
+
+		if ( self::OK === $valid ) {
+			$this->status = self::OK;
+
+			return $this->status;
+		}
+
+		if ( self::ERROR === $valid ) {
+			$this->status  = self::ERROR;
+			$this->sitemap = '';
+
+			return $this->status;
+		}
+
+		$this->status = self::UNKNOWN;
+
+		return $this->status;
+	}
+
+	/**
+	 * Is the sitemap usable.
+	 *
+	 * @param bool|null $usable If set; adjust status accordingly.
+	 *
+	 * @return bool True if usable, False if bad or unknown.
+	 */
+	public function is_usable( $usable = null ) {
+		if ( ! is_null( $usable ) && is_bool( $usable ) ) {
+			$this->status = ( $usable ) ? self::OK : self::ERROR;
+		}
+
+		return self::OK === $this->status;
+	}
+
+	/**
+	 * Get the XML content of the sitemap.
+	 *
+	 * @return string The content of the sitemap.
+	 */
+	public function get_sitemap() {
+
+		return $this->sitemap;
+	}
+
+	/**
+	 * Get the usability of the sitemap.
```

Comment:
I would expect that the function `is_usable()` is called right now, but instead the status is returned, not the usability (as stated in `is_usable()`). Perhaps it's better to change the comment to: Get the status of the sitemap.